airdrop-machine/sources/airdrop_machine.move

```diff
@@ -0,0 +1,289 @@
+        assert_owner(signer::address_of(admin), collection);
+        mint_impl(admin, collection, name, description, uri, recipient_addr)
+    }
+
+    public fun mint_impl(
+        _minter: &signer,
+        collection: Object<Collection>,
+        name: String,
+        description: String,
+        uri: String,
+        recipient_addr: address,
+    ): Object<Token> acquires CollectionConfig {
+        assert!(minting_started(collection), error::permission_denied(EMINTING_HAS_NOT_STARTED));
+
+        let collection_signer = collection_signer(collection);
+        
+        let constructor_ref = &token::create_numbered_token(
+            &collection_signer,
+            collection::name(collection),
+            description,
+            name,
+            utf8(b""), // name_with_index_suffix 
+            royalty::get(collection),
+            uri,
+        );
+
+        token_components::create_refs(constructor_ref);
+        transfer_token::transfer(&collection_signer, recipient_addr, constructor_ref);
+        object::object_from_constructor_ref(constructor_ref)
+    }
+
+    public fun create_collection_impl(
+        admin: &signer,
+        description: String,
+        name: String,
+        uri: String,
+        mutable_collection_metadata: bool,
+        mutable_token_metadata: bool,
+        tokens_burnable_by_collection_owner: bool,
+        tokens_transferrable_by_collection_owner: bool,
+        max_supply: Option<u64>,
+        royalty_numerator: Option<u64>,
+        royalty_denominator: Option<u64>,
+    ): Object<Collection> {
+        let admin_addr = signer::address_of(admin);
+        let object_constructor_ref = &object::create_object(admin_addr);
+        let object_signer = object::generate_signer(object_constructor_ref);
+        let royalty = royalty(&mut royalty_numerator, &mut royalty_denominator, admin_addr);
+
+        let constructor_ref = if (option::is_some(&max_supply)) {
+            collection::create_fixed_collection(
+                &object_signer,
+                description,
+                option::extract(&mut max_supply),
+                name,
+                royalty,
+                uri,
+            )
+        } else {
+            collection::create_unlimited_collection(
+                &object_signer,
+                description,
+                name,
+                royalty,
+                uri,
+            )
+        };
+
+        collection_components::create_refs_and_properties(&constructor_ref);
+        let collection = object::object_from_constructor_ref(&constructor_ref);
+        configure_collection_and_token_properties(
+            &object_signer,
+            collection,
+            mutable_collection_metadata,
+            mutable_token_metadata,
+            tokens_burnable_by_collection_owner,
+            tokens_transferrable_by_collection_owner,
+        );
+
+        let object_signer = object::generate_signer(&constructor_ref);
```

Comment:
name this `collection_object_signer` as we already have an `object_signer` above.